nvi-test/src/main/java/no/sikt/nva/nvi/test/ExpandedPublication.java

```diff
@@ -0,0 +1,240 @@
+package no.sikt.nva.nvi.test;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static no.sikt.nva.nvi.test.TestConstants.EN_FIELD;
+import static no.sikt.nva.nvi.test.TestConstants.HARDCODED_ENGLISH_LABEL;
+import static no.sikt.nva.nvi.test.TestConstants.HARDCODED_NORWEGIAN_LABEL;
+import static no.sikt.nva.nvi.test.TestConstants.IDENTIFIER_FIELD;
+import static no.sikt.nva.nvi.test.TestConstants.ID_FIELD;
+import static no.sikt.nva.nvi.test.TestConstants.LABELS_FIELD;
+import static no.sikt.nva.nvi.test.TestConstants.NB_FIELD;
+import static no.sikt.nva.nvi.test.TestConstants.ONE;
+import static no.sikt.nva.nvi.test.TestConstants.PAGES_FIELD;
+import static no.sikt.nva.nvi.test.TestConstants.STATUS_FIELD;
+import static no.sikt.nva.nvi.test.TestConstants.TYPE_FIELD;
+import static no.sikt.nva.nvi.test.TestUtils.generatePublicationId;
+import static no.unit.nva.testutils.RandomDataGenerator.objectMapper;
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.ioutils.IoUtils;
+
+/**
+ * This is intended to create a JSON representation of an expanded resource, mirroring the expected input documents from
+ * `nva-publication-api`.
+ */
+@JacocoGenerated
+public record ExpandedPublication(URI id, UUID identifier, String mainTitle, String issn, String publicationContextType,
+                                  List<ExpandedPublicationChannel> publicationChannels,
+                                  ExpandedPublicationDate publicationDate, String instanceType, String abstractText,
+                                  String language, String status, List<ExpandedContributor> contributors) {
+
+    private static final String TEMPLATE_JSON_PATH = "template_publication.json";
+    private static final String PUBLICATION_CHANNELS_MUST_NOT_BE_EMPTY = "Publication channels must not be empty";
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Builder from(ExpandedPublication other) {
+        return new Builder(other);
+    }
+
+    public String toJsonString() {
+        return toJsonNode().toString();
+    }
+
+    public JsonNode toJsonNode() {
+        var root = objectMapper.createObjectNode();
+        root.set("body", createExpandedResource());
+        return root;
+    }
+
+    private static ObjectNode createAndPopulatePublicationInstance(String type) {
+        var publicationInstance = objectMapper.createObjectNode();
+        publicationInstance.put(TYPE_FIELD, type);
+
+        switch (type) {
+            case "AcademicArticle", "AcademicLiteratureReview", "AcademicChapter" -> {
```

Comment:
These types should also be defined as constants somewhere in project, maybe commons?

Reply:
Agreed, but lets do it in a separate PR. I want to fix the circular dependencies here before we add more imports across sub-projects.